exp/api/v1beta1/azuremanagedmachinepool_webhook_test.go

```diff
@@ -436,10 +436,31 @@ func TestAzureManagedMachinePoolUpdatingWebhook(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
+			name: "Can't add a node label that begins with kubernetes.azure.com",
+			new: &AzureManagedMachinePool{
+				Spec: AzureManagedMachinePoolSpec{
+					NodeLabels: map[string]string{
+						"foo":                                   "bar",
+						"kubernetes.azure.com/scalesetpriority": "spot",
+					},
+				},
+			},
+			old: &AzureManagedMachinePool{
+				Spec: AzureManagedMachinePoolSpec{
+					NodeLabels: map[string]string{
+						"foo": "bar",
+					},
+				},
+			},
+			wantErr: true,
+		},
 	}
 	var client client.Client
 	for _, tc := range tests {
+		tc := tc
```

Comment:
Two points:

1. It's correct that we do this as a best-practice to avoid the non-deterministic UT outcomes that result from not doing this when we need to.
2. Doing it all the time as a rote practice tends to make the developer forget the exact edge case root cause that makes this pattern a necessary thing and it becomes a kind of incantation.

🤷